### ISSUE TITLE
Clicking on a button when the button is offscreen

### ISSUE THREAD

**Issue by sam22blr** [TestStack/White#317]
When the button is offscreen, the click doesn't work nor it throws any exception, is there a method to make the button onscreen before clicking on it?

**Comment by maxinfet** [comment]
There should be a parameter to see if its visible but you will then need to do something to bring it into view like scrolling  a scroll viewer or moving a window into view.

**Comment by sam22blr** [comment]
Was thinking if there is an easier way to do this like the make visible method in coded ui.

**Comment by dpisanu** [comment]
Nope, there is no "BringToFocus" kind of functionality out of the box.
This is something you will need to do yourself.

Since a Button is usually in some sort of Window and not free floating by itself, you can call Window.Focus() on the Parent Window Container.

Since it is a Button, you could see if playing with the Invoke Pattern might just do the trick even if the Button is
```javascript
Button.IsOffScreen == true
``` 
or 
```javascript
Visible == false
```

Try this.
```javascript
var pattern = Button.AutomationElement.GetCurrentPattern(System.Windows.Automation.InvokePattern.Pattern);
(pattern as System.Windows.Automation.InvokePattern).Invoke();
```

**Comment by pottabathini** [comment]
@dpisanu perfect answer 👍
